src/millie.nim

```diff
@@ -10,46 +10,45 @@ const units = @[
   "P", # quadrillion
 ]
 
-## Converts big numbers to what's pleasant to see (an adorable, little girl, perhaps?)
+
 proc millify*(
-    value: int64,
+    value: SomeNumber,
     precision: int = 1,
     lowercase: bool = false,
     space: bool = false,
   ): string =
+  ## Converts big numbers to what's pleasant to see (an adorable, little girl, perhaps?)
   
-  # validate if `vaue` can be parsed or not
-  var val: SomeNumber = parseNum(value)
+  # preallocate the resulting string
+  result = newStringOfCap(10)
 
-  # `prefix` will be used later when assembling
-  let prefix: string = if val < 0: "-" else: ""
+  var val = float(value)
 
+  # add a negative prefix if needed
+  if val < 0: result.add '-'
+  
   # only work with positive numbers for easiness
   val = val.abs()
 
   # keep dividing the input value by the digit grouping base (1000)
   # until the number and the unit index are found.
-  var unitIndex: int = 0
-  proc disposableFunc(res: float) =
+  var unitIndex = 0
+  divideTillEnd(val):
     val = res
     unitIndex += 1
-  divideTillEnd(val, disposableFunc)
 
   # The number is too big for Millie to handle. So,
   # Millie just returns it as is.
   if len(units) <= unitIndex: return $value
 
   # round the decimal number up to the desired precision
-  var rounded: SomeNumber = roundTo(val, precision)
-  var formatted: string
-  if rounded.ceil() == rounded: formatted = $rounded.ceil().float.int else: formatted = $rounded
-
-  # get the appropriate unit for the value
-  let unit: string = if len(units) > unitIndex: units[unitIndex] else: ""
```

Comment:
The check for `if len(units) > unitIndex` is redundant since we already returned the original value if `if len(units) <= unitIndex:` in the code before